### PR TITLE
Upgraded to TypeScript 3.0.1.

### PR DESCRIPTION
**Pull request by christopherstock**
Upgraded to TypeScript 3.0.1 and Fixed type errors in interface VRLayer by enriching the possible types
 for the subsequent properties 'leftBounds' and 'rightBounds'.

### CODE REVIEW

**Comment by deltakosh** [comment]
All tests seem to failed:(

File: dist/preview release/loaders/babylonjs.loaders.js [hunk 1]
```diff
@@ -2956,7 +2956,7 @@ var BABYLON;
                     (sampler.minFilter === GLTF1.ETextureFilterType.LINEAR_MIPMAP_NEAREST) ||
                     (sampler.minFilter === GLTF1.ETextureFilterType.LINEAR_MIPMAP_LINEAR);
                 var samplingMode = BABYLON.Texture.BILINEAR_SAMPLINGMODE;
-                var blob = new Blob([buffer]);
```
Comment (deltakosh) [inline dist/preview release/loaders/babylonjs.loaders.js]: I would prefer use [buffer!]

**Comment by christopherstock** [comment]
Wouldn't it be better if the "dist" folder is under "git ignore"?
I'm trying to revert it from the original repository.

**Comment by deltakosh** [comment]
Unfortunately in this case I would have to fight to generate it :)

**Comment by deltakosh** [comment]
No worry I'll clean it here ;)

**Comment by christopherstock** [comment]
woah! thank you so much for your cooperation! 💃 
Have a great weekend and enjoy yourself! :D 

Sincerely yours
Christopher

**Comment by deltakosh** [comment]
Thank you for your PR

**Comment by 0x53A** [comment]
Can you please publish this?

I did 

1) ``create-react-app my-app --scripts-version=react-scripts-ts``

2) ``yarn add babylonjs``

and get this error:


image

I "fixed" it manually by replacing this block ...

```ts
interface VRLayer {
    leftBounds?: number[] | null;
    rightBounds?: number[] | null;
    source?: HTMLCanvasElement | null;
}
```

with this block:

```ts
interface VRLayer {
    leftBounds?: number[] | Float32Array | null;
    rightBounds?: number[] | Float32Array | null;
    source?: HTMLCanvasElement | null;
}
```

but it kinda sucks that the current typescript version doesn't seem to be compatible with the current babylonjs version (3.2.0).

**Comment by sebavan** [comment]
Can you use the npm preview version of 3.3 ?

**Comment by 0x53A** [comment]
``3.3.0-beta.1`` works

**Comment by sebavan** [comment]
Sweet :-)

**Pull request merged**